### DISCUSSION TITLE
Adding collaborative functionalities to SlateEditor

### DISCUSSION THREAD

**Discussion by KarimAbouelNaga** [ianstormtaylor/slate#4399] [Q&A]
Hi there!

I have been using Slate to implement a real-time collaborative text editor using web sockets. I want to give last polishments and plan to add some usual collaborative properties such as seeing the position of another user.

Does anyone have experience or made something familiar in that direction?

Best

Karim

**Comment by williamstein** [comment]
> Does anyone have experience or made something familiar in that direction?

I made a real-time collaborative text editor for Markdown documents (with support for latex-style math, checkboxes, colors, fonts, table of contents, resizable images, codemirror code editing, etc.) using Slate.  It provides the WYSIWYG editor for Markdown in https://cocalc.com.

**Reply by williamstein** [reply]
The source code is here:  https://github.com/sagemathinc/cocalc/tree/master/src/smc-webapp/editors/slate
